src/foam/nanos/auth/PasswordPolicy.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'PasswordPolicy',
+
+  implements: [
+    'foam.nanos.auth.EnabledAware'
+  ],
+
+  documentation: 'A Password Policy for a Group.',
+  
+  constants: [
+    {
+      name: 'MIN_PASSWORD_LENGTH',
+      type: 'Long',
+      value: 6
+    }
+  ],
+
+  properties: [
+    {
+      class: 'Boolean',
+      name: 'enabled'
+    },
+    {
+      class: 'Long',
+      name: 'minLength',
+      documentation: 'Minimum length of the password.',
+      min: 6,
+      value: 6
+    },
+    {
+      class: 'Long',
+      name: 'preventHistoricPasswordCount',
+      min: 0,
+      max: 10,
+      documentation: `
+        User passwords are hashed and the hash is kept for historical record up to a maximum of 10 previous entries per user. 
+        Setting preventHistoricalPasswordCount to a value greater than zero will enable a check on that many previous password entries
+        when resetting the users passwords to prevent them from using the same password again.
+        `
+    },
+  ],
+
+  javaImports: [
+    'foam.nanos.auth.User',
+    'foam.util.Password',
+    'foam.util.SafetyUtil',
+    'java.util.regex.Pattern'
+  ],
+
+  methods: [
+    {
+      name: 'validate',
+      async: true,
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        },
+        {
+          name: 'user',
+          type: 'User'
+        },
+        {
+          name: 'potentialPassword',
+          javaType: 'String'
+        }
+      ],
+      javaCode: `
+        // check if this policy is enabled
+        if ( ! this.getEnabled() ) {
+          return; // password is valid by default
+        }
+
+        // check minimum length
+        int minLength = (int)Math.max(this.getMinLength(), MIN_PASSWORD_LENGTH);
+        String minLengthRegex = "^.{" + minLength + ",}$";
```

Comment:
Wouldn't it be simpler just to check `potentialPassword.length >= minLength`?

Reply:
The `SafetyUtil.isEmpty(potentialPassword)` you're doing anyway catches nulls, so it'll be safe to do the length check after that one.

It was probably a more complicated regex before. Now that it's simply a length check it's simpler just to check the length of the string directly.